stats/cloud/collector.go

```diff
@@ -441,15 +443,28 @@ func (c *Collector) flushHTTPTrails() {
 	c.aggrBuckets = map[int64]aggregationBucket{}
 	c.bufferSamples = append(c.bufferSamples, newSamples...)
 }
+
+func (c *Collector) shouldStopSentMetric(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	return strings.HasPrefix(err.Error(), "(403/E4)")
```

Comment:
Would it make sense to make "(403/E4)" a global constant?

It might be good to document what it refers to or where it comes from as well. I'm assuming 403 is HTTP 403, but not sure what E4 stands for.

Also, is "limit reached" the only scenario we wish to stop sending metrics on? I can't imagine others, but maybe we should consider and handle them here as well.

Reply:
@imiric This is the only place where it check for `"(403/E4)"`, that's why I don't make it a global constant. You can read `errors.go` to understand more, I construct this base on original description in issue.

@na-- @MStoykov can you provide more context?

Reply:
> Also, is "limit reached" the only scenario we wish to stop sending metrics on? I can't imagine others, but maybe we should consider and handle them here as well.

Good question. There are probably other ones, but let's focus on this one for now. I think that the way @cuonglm implemented the check in a helper `shouldStopSendingMetrics()` method would give us enough flexibility to add other conditions in the future. 

However, the stringy way we're doing the comparison is probably not optimal. We have a very detailed type for structured errors by the backend, so we should probably use that: https://github.com/loadimpact/k6/blob/2326f048e94159d4ec83a0d08be10003e3e13250/stats/cloud/errors.go#L37-L46

Reply:
@na-- I read that struct too, but it does hide the Response body, so we can't check for status 403. Since when we does have implement error format string, then checking string is sufficient enough IMHO.

Reply:
> but it does hide the Response body

This needs to be verified, but I see that there's a `Response *http.Response` in that struct - won't that be sufficient to check for the response status?

Reply:
@na-- Ah right, we ignore when serialize to client, not when we deserialize from api server.